src/dom/element.rs

```diff
@@ -481,7 +483,9 @@ where
                     let (expr_id, content_id) = if let Some(placeholder) = next_placeholder {
                         (placeholder, None)
                     } else {
-                        create_placeholder(results, &temp_path, index, "")
+                        let placeholder= create_placeholder(results, &temp_path, index, "");
```

Comment:
I'd expect create_placeholder to be the thing that needs to increment the index, no?